pkg/plugin/aggregation/aggregator.go

```diff
@@ -201,54 +201,76 @@ func (a *Aggregator) handleResult(result *plugin.Result) error {
 		a.resultEvents <- result
 	}()
 
-	var resultsDir string
-	var outFile *os.File
-	var err error
-
 	if result.MimeType == gzipMimeType {
-		resultsDir = path.Join(a.OutputDir, result.Path())
-		// Gzip gets a temp file
-		outFile, err = ioutil.TempFile("", "aggregator-gzip")
-		if err != nil {
-			return errors.Wrapf(err, "could not make tempfile for gzip")
-		}
-
-		defer outFile.Close()
-		// Remove the temp file when we're done
-		defer os.Remove(outFile.Name())
-	} else {
-		// Create the output directory for the result.  Will be of the
-		// form .../plugins/:results_type/:node.json (for DaemonSet plugins) or
-		// .../plugins/:results_type.json (for Job plugins)
-		resultsFile := path.Join(a.OutputDir, result.Path())
-		resultsDir = path.Dir(resultsFile)
-		outFile, err = os.Create(resultsFile)
-		if err != nil {
-			return errors.Wrapf(err, "couldn't create results file %v", resultsFile)
-		}
-		defer outFile.Close()
+		return a.handleArchiveResult(result)
 	}
 
-	logrus.Infof("Creating directory %v", resultsDir)
+	// Create the output directory for the result.  Will be of the
+	// form .../plugins/:results_type/:node.json (for DaemonSet plugins) or
+	// .../plugins/:results_type.json (for Job plugins)
+	resultsFile := path.Join(a.OutputDir, result.Path())
+	resultsDir := path.Dir(resultsFile)
+
 	if err := os.MkdirAll(resultsDir, 0755); err != nil {
-		errlog.LogError(err)
+		errors.Wrapf(err, "couldn't create directory %v", resultsDir)
 		return err
 	}
 
+	outFile, err := os.Create(resultsFile)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't create results file %v", resultsFile)
+	}
+	defer outFile.Close()
+
 	if _, err = io.Copy(outFile, result.Body); err != nil {
 		err = errors.Wrapf(err, "could not write body to file %v", outFile.Name())
 		return err
 	}
 
-	if result.MimeType == gzipMimeType {
-		// extract to the working directory
-		err = tarx.Extract(outFile.Name(), resultsDir, &tarx.ExtractOptions{})
+	return nil
+
+}
+
+func (a *Aggregator) handleArchiveResult(result *plugin.Result) error {
```

Comment:
this could be a little more general and not be a method on the aggregator. What do you think about a function that looked like `Extract(readCloser io.ReadCloser, dir string) error`? I was thinking about that as a standalone library, but would be fine inside sonobuoy as well.

Reply:
If I wanted to make this general, I'd need to handle a bunch more stuff. I'm not setting uid/guid, not handling symlinks/hardlinks, all that stuff. So I don't want people to see this library and think it's up for general use.